### PR TITLE
Feature/chapters index page

### PR DESCRIPTION
**Pull request by yihyang**
Issue: https://github.com/codebar/planner/issues/738

Screenshot:

Screenshot 2020-05-17 at 11 31 30 PM

### CODE REVIEW

**Comment by matyikriszta** [comment]
@yihyang thanks for the PR, please give me a few days to review it, I'll try to get to it this week

File: spec/features/admin/chapters_spec.rb [hunk 1]
```diff
@@ -14,6 +14,72 @@
...
+    background { login_as_admin(member) }
+
+    scenario "an admin can view all chapters and it's information" do
+      visit admin_chapters_path
+
+      assert_chapters_exist_on_page(active_chapters)
+      assert_chapters_exist_on_page(inactive_chapters)
+    end
+
+    scenario 'an admin can filter and view only active chapters' do
```
[the remark this answers is not in the document]
Comment (yihyang) [reply spec/features/admin/chapters_spec.rb]: Hi @notapatch thanks for the suggestions, I have updated it in https://github.com/codebar/planner/pull/1069/commits/91c732db8913a3cc47c6ddbf363486744793194c

**Comment by yihyang** [comment]
Hi @matyikriszta @notapatch , sorry for the late reply. I hope that everyone is safe during this times. I have updated the PR as per suggestion, please have a look, thanks!

File: spec/support/wait_for_ajax.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+module WaitForAjax
```
Comment (yihyang) [inline spec/support/wait_for_ajax.rb:1]: copied this from https://thoughtbot.com/blog/automatically-wait-for-ajax-with-capybara

**Comment by notapatch** [comment]
Hi @yihyang 

@despo, the project owner, is active but I don't know the project's priorities as codebar has had to pivot to online workshops which has led to a lot of code changes. I hope she has the time.🤞

File: app/controllers/admin/chapters_controller.rb [hunk 3]
```diff
@@ -59,13 +66,29 @@ def members
     render plain: @emails
   end
 
+  def update_active
+    authorize @chapter
+
+    @chapter.update(update_active_params)
+
+    head :ok
+  end
```
Comment (despo) [inline app/controllers/admin/chapters_controller.rb:75]: Can we stick to standard rails CRUD operations and reuse `update` to achieve this instead. The `active` param can just be part of chapter_params and the existing `update` method could be used to achieve the same thing.

File: app/views/admin/chapters/index.html.haml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          %p= chapter.name
+        %p= chapter.city
+        %p= chapter.slug
+      .large-4.columns
+        %p= chapter.email
+        %p= chapter.twitter
+      .large-1.columns
+        = check_box_tag :chapter_active, 'true', chapter.active?, class: 'chapter_active', data: { 'chapter-id': chapter.id, }
+
+- content_for :page_footer do
```
Comment (despo) [inline app/views/admin/chapters/index.html.haml:32]: I don't mind having javascript available as well, however it should be optional and not the default. Also, a confirmation button to verify disabling chapters before any actions take place.

File: app/views/admin/chapters/index.html.haml [hunk 5]
```diff
@@ -0,0 +1,45 @@
+%section#banner
+  .row
```
Comment (despo) [inline app/views/admin/chapters/index.html.haml:2]: a `.row` in foundation should contain column elements, so here there should be a `large-12.columns` to achieve correct alignment.

File: app/views/admin/chapters/index.html.haml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    %h1 Chapters
+    = link_to t('.view_all_chapters'), admin_chapters_path
+    &nbsp;
+    = link_to t('.view_active_chapters'), admin_chapters_path(active: true)
+    &nbsp;
+    = link_to t('.view_inactive_chapters'), admin_chapters_path(active: false)
+    .row
+      %hr
```
Comment (despo) [inline app/views/admin/chapters/index.html.haml:10]: similar to the comment above, as a row is made up of column elements, `hr` should be within a ` .large-12.columns`

File: config/routes.rb [hunk 7]
```diff
@@ -119,6 +119,10 @@
     resources :member_notes, only: [:create]
 
     resources :chapters, only: %i[index new create show edit update] do
+      member do
```
Comment (despo) [inline config/routes.rb:122]: this can be removed as per comments added above

File: spec/features/admin/chapters_spec.rb [hunk 8]
```diff
@@ -14,6 +14,70 @@
     end
   end
 
+  context '#index' do
+    let!(:active_chapters) { Fabricate.times(3, :active_chapter) }
+    let!(:inactive_chapters) { Fabricate.times(3, :inactive_chapter) }
+
+    background { login_as_admin(member) }
+
+    scenario "an admin can view all chapters and it's information" do
```
Comment (despo) [inline spec/features/admin/chapters_spec.rb:23]: could you please update to `...all chapter and their information`

**Pull request closed without merging**